src/components/scenes/GettingStartedScene.tsx

```diff
@@ -80,7 +80,7 @@ const REMOTE_CONFIG_DEFAULT = {
 
 export const GettingStartedScene = (props: Props) => {
   const { navigation } = props
-  const localUsersLength = useSelector(state => state.core.context.localUsers.length)
+  const hasLocalUsers = useSelector(state => state.core.context.localUsers.length > 0)
```

Comment:
A "more correct" solution would be:

```ts
const context= useSelector(state => state.core.context)
const localUsers = useWatch(context, 'localUsers')
const const hasLocalUsers = localUsers.length >0
```

Doing it all in the selector means it won't necessarily update correctly, especially since this is deletion-related.

Reply:
If `fn` in `useSelector(fn)` is invoked on changes to the redux state tree and `===` compared to the previous selector's return value before triggering a re-render for `GettingStartedScene`, then making the change you proposed will cause more unnecessary re-renders unless I am misunderstanding redux. Please explain if I am misunderstanding.

Reply:
The first line will never cause a re-render, because the context will remain `===` the same for the whole lifetime of the app. The only exception is if the WebView reloads, in which case we should probably render anyhow.

The second line will render any time users are added or removed, which we do want. The `useSelector` function has *no idea* when the core is going to experience these changes, so we need to subscribe to the core itself. Otherwise, if a user gets deleted without a corresponding `dispatch` (which may happen as we rely on Redux less and less), the component will never see the change and never update.

Finally, the third line implements your fix. By creating an intermediate variable, you can precisely control when the `useEffect` gets triggered based on `===` changes.

Reply:
Ah, I see now that you used `useWatch` and I missed that part. Got it. Thanks for the explanation.